lib/monitoring/middleware/conjur_collector.rb

```diff
@@ -0,0 +1,65 @@
+require 'benchmark'
+require ::File.expand_path('../../pub_sub.rb', __FILE__)
+
+module Monitoring
+  module Middleware
+    class ConjurCollector
+      attr_reader :app, :registry, :pubsub
+
+      def initialize(app, options = {})
+        @app = app
+        @registry = options[:registry]
+        @pubsub = options[:pubsub]
```

Comment:
I don't think there is a need to pass in a pubsub object here since in the rest of the class it looks like you are making static method calls on the Monitoring::PubSub module. Plus it would make the pubsub mechanism a bit more intrusive if we needed to rely on dependency injection for any modules/classes which need to be able to publish events.